### ISSUE TITLE
Data is stored from client and viewable from redis-cli but returns undefined in console.log

### ISSUE THREAD

**Issue by xxkenta** [AkashBabu/redis-json#7]
I've copied the sample code with a few adjustments, and am able to see the data from SET getting into redis from redis-cli, however I cannot get the data back into node to print to console.
```
const redis = require('redis');
const JSONCache = require('redis-json');
const redis_port = process.env.REDIS_PORT || 6379

const client = redis.createClient({host: "127.0.0.1", port: redis_port})
const jsonCache = new JSONCache(client, {prefix: 'cache:'});

const user = {
    name: "kenta",
    age: 22,
    address: {
        street: "2840 Bay Colony Lane",
        city: "Lexington",
        state: "KY"
    },
    cars: ['300zx', 'WD21']
}

async function setCache(cache, object) {
    await cache.set('0', object)
    let response = await cache.get('0')
    console.log(response)
}

setCache(jsonCache, user)
```

However, when I run this the response is always `undefined`

Yet in redis-cli:

```
127.0.0.1:6379> hgetall cache:0
 1) "name"
 2) "kenta"
 3) "age"
 4) "22"
 5) "address.street"
 6) "ABCD 1234"
 7) "address.city"
 8) "NYC"
 9) "address.state"
10) "NY"
11) "cars.0"
12) "300zx"
13) "cars.1"
14) "WD21"
15) "__jc_root__"
16) "0"
```

**Comment by xxkenta** [comment]
Another bit to note, if I run in monitor mode on redis-cli, and run the same bit of code, it shows:

```
127.0.0.1:6379> monitor
OK
1585668218.091359 [0 127.0.0.1:54366] "info"
1585668218.094267 [0 127.0.0.1:54366] "hmset" "jc:0" "name" "kenta" "age" "22" "address.street" "2840 Bay Colony Lane" "address.city" "Lexington" "address.state" "KY" "cars.0" "300zx" "cars.1" "WD21" "__jc_root__" "0"
1585668218.094294 [0 127.0.0.1:54366] "hgetall" "jc:0"
```

So clearly it is calling `hgetall` but still showing results as undefined.

**Comment by xxkenta** [comment]
Another bit to note, if I run in monitor mode on redis-cli, and run the same bit of code, it shows:

```
127.0.0.1:6379> monitor
OK
1585668218.091359 [0 127.0.0.1:54366] "info"
1585668218.094267 [0 127.0.0.1:54366] "hmset" "jc:0" "name" "kenta" "age" "22" "address.street" "ABCD 1234" "address.city" "NYC" "address.state" "NY" "cars.0" "300zx" "cars.1" "WD21" "__jc_root__" "0"
1585668218.094294 [0 127.0.0.1:54366] "hgetall" "jc:0"
```

So clearly it is calling `hgetall` but still showing results as undefined.

**Comment by AkashBabu** [comment]
oh! this behaviour is unexpected. Let me check and revert back to you.
Can you also mention the version of the library (redis-json) being used ?

Thanks for reporting!

**Comment by xxkenta** [comment]
> oh! this behaviour is unexpected. Let me check and revert back to you.
> Can you also mention the version of the library (redis-json) being used ?
> 
> Thanks for reporting!

Yes! I am using version 3.2.0. Thanks for looking into this!
One thing to note, I'm unsure if it is related. I am using windows, with Redis running on the WSL (Ubuntu 18.04 LTS). Let me know if I can provide anymore information to help.

**Comment by AkashBabu** [comment]
@xxkenta this issue has been fixed in v3.2.1

Just FYI, this issue was because we had all the test cases written around [`ioredis`](https://www.npmjs.com/package/ioredis) which supports both callbacks and promises, but whereas `redis` library support only callbacks because of which the results were incorrect.
Now we are promisifying both the clients and hence the issue has been resolved.

**Comment by xxkenta** [comment]
> @xxkenta this issue has been fixed in v3.2.1
> 
> Just FYI, this issue was because we had all the test cases written around [`ioredis`](https://www.npmjs.com/package/ioredis) which supports both callbacks and promises, but whereas `redis` library support only callbacks because of which the results were incorrect.
> Now we are promisifying both the clients and hence the issue has been resolved.

Thank you very much for your prompt resolution!
